templates/xpress/main.py

```diff
@@ -0,0 +1,126 @@
+"""
+Template for working with FICO Xpress.
+"""
+
+import argparse
+import json
+import sys
+from typing import Any, Dict
+
+import xpress as xp
+
+
+def main() -> None:
+    """Entry point for the template."""
+
+    parser = argparse.ArgumentParser(description="Solve problems with Xpress.")
+    parser.add_argument(
+        "-input",
+        default="",
+        help="Path to input file. Default is stdin.",
+    )
+    parser.add_argument(
+        "-output",
+        default="",
+        help="Path to output file. Default is stdout.",
+    )
+    parser.add_argument(
+        "-duration",
+        default=30,
+        help="Max runtime duration (in seconds). Default is 30.",
+        type=int,
+    )
+    args = parser.parse_args()
+
+    # Read input data, solve the problem and write the solution.
+    input_data = read_input(args.input)
+    solution = solve(input_data, args.duration)
+    write_output(args.output, solution)
+
+
+def solve(input_data: Dict[str, Any], duration: int) -> Dict[str, Any]:
+    """Solves the given problem and returns the solution."""
+
+    # Creates the problem.
+    problem = xp.problem()
+    problem.setControl("timelimit", duration)
+
+    # Initializes the linear sums.
+    weights = 0.0
+    values = 0.0
+
+    # Creates the decision variables and adds them to the linear sums.
+    items = []
+    for item in input_data["items"]:
+        item_variable = xp.var(vartype=xp.binary, name=item["item_id"])
+        problem.addVariable(item_variable)
+        items.append({"item": item, "variable": item_variable})
+        weights += item_variable * item["weight"]
+        values += item_variable * item["value"]
+
+    # This constraint ensures the weight capacity of the knapsack will not be
+    # exceeded.
+    problem.addConstraint(weights <= input_data["weight_capacity"])
+
+    # Sets the objective function: maximize the value of the chosen items.
+    problem.setObjective(values, sense=xp.maximize)
+
+    # Solves the problem.
+    _, status = problem.optimize()
+
+    # Determines which items were chosen.
+    chosen_items = []
+    for item in items:
+        if problem.getSolution(item["variable"]) > 0.9:
+            chosen_items.append(item["item"])
+
+    # Creates the statistics.
+    statistics = {
+        "result": {
+            "custom": {
+                "constraints": problem.getAttrib("rows"),
+                "provider": "xpress",
+                "status": status,
```

Comment:
I compared our HiGHS solver against SCIP via OR-Tools and this clashed a bit. It's probably fine, but we could think about translating this to the same type or even enum (`unbounded`, `optimal`, etc.). The biggest problem was that ortools returns an int here and we return a string like mentioned before. I am not sure what xpress python uses.
It's probably too complicated, but maybe we should at least make them all strings so that experiments aren't confused by the column? :shrug: 
Just something I noticed.

Reply:
I'll align with our SDK to print a string.